coldfront/plugins/qumulo/tests/utils/test_eib_billing.py

```diff
@@ -0,0 +1,352 @@
+                    ) AS most_recent_haau
+                    ON aau.allocation_attribute_id = most_recent_haau.allocation_attribute_id
+                JOIN allocation_allocationattributetype aat
+                    ON aat.id = aa.allocation_attribute_type_id
+                JOIN (
+                    SELECT aa.allocation_id, aa.value storage_filesystem_path
+                        FROM allocation_allocationattribute aa
+                        JOIN allocation_allocationattributetype aat
+                           ON aa.allocation_attribute_type_id = aat.id
+                        WHERE aat.name = 'storage_filesystem_path'
+                    ) AS storage_filesystem_path
+                    ON aa.allocation_id = storage_filesystem_path.allocation_id
+                ORDER BY id DESC, history_timestamp DESC;
+            """
+            )
+            rows = cursor.fetchall()
+
+        for row in rows:
+            # Confirm the new usage is not 0
+            print(row)
+            self.assertNotEqual(float(row[3]) - 0, 0)
+
+        # Confirm the status of the allocation is Active
+        allocations = Allocation.objects.filter(resources__name="Storage2").exclude(
+            status__name__in=[
+                "Active",
+            ]
+        )
+        self.assertEqual(len(list(allocations)), 0)
+
+        eib_billing = EIBBilling(datetime.now(timezone.utc).strftime("%Y-%m-%d"))
+        eib_billing.generate_monthly_billing_report()
+
+        filename = eib_billing.get_filename()
+        self.assertFalse(re.search("RIS-%s-storage2-active-billing.csv", filename))
+        self.assertTrue(
+            re.search("RIS-[A-Za-z]+-storage2-active-billing.csv", filename)
+        )
+        self.assertTrue(os.path.exists(filename))
+        os.system(f"ls -l {filename}")
+
+        with open(filename) as csvreport:
+            data = list(csv.reader(csvreport))
+
+        # Confirm the billing amount for 5 unit of Consumption cost model is $65
+        # hardcoded
+        billing_amount = float(data[len(data) - 1][19])
+        self.assertEqual(billing_amount - 65.0, 0)
+
+        os.remove(filename)
+
+    @patch("coldfront.plugins.qumulo.tasks.QumuloAPI")
+    def test_create_allocations_ingest_usages_generate_billing_report(
+        self, qumulo_api_mock: MagicMock
+    ) -> None:
+        qumulo_api = MagicMock()
+        qumulo_api.get_all_quotas_with_usage.return_value = mock_get_quotas()
+        qumulo_api_mock.return_value = qumulo_api
+
+        quota_service_rate_categories = mock_get_quota_service_rate_categories()
+
+        for quota, service_rate_category in quota_service_rate_categories:
+            create_allocation(
+                project=self.project,
+                user=self.user,
+                form_data=construct_allocation_form_data(quota, service_rate_category),
+            )
+
+        # Update the status of the created allocations from Pending to Active
```

Comment:
allocations using the `create_allocation` method from `mock_data` are already active

Reply:
No I think it's still in the "Pending" status as far as I tested.

Reply:
Ahh, yes.  I don't know why I misread that.